python/src/main/java/org/apache/zeppelin/python/PythonInterpreter.java

```diff
@@ -54,8 +55,9 @@
 
   private Integer port;
   private GatewayServer gatewayServer;
-  private Boolean py4J = false;
+  private Boolean py4JisInstalled = false;
   private InterpreterContext context;
+  private Pattern errorInLastLine = Pattern.compile(".*(Error|Exception): .*$");
```

Comment:
Good points, thank you @felixcheung !

There is definitely room for improvement - this is more like a foundation of this feature, as previous impl always returned `SUCCESS`.

Your example of `print("there is no Error: ok")` will indeed trigger the the false positive ERROR status.

I have filed [ZEPPELIN-1114](https://issues.apache.org/jira/browse/ZEPPELIN-1114) and logged this feedback there. I plan to take care of it in a subsequent PRs.